src/Orchard.Data/ServiceCollectionExtensions.cs

```diff
@@ -25,6 +27,8 @@ public static IServiceCollection AddDataAccess(this IServiceCollection services)
             services.TryAddDataProvider(name: "Sql Server", value: "SqlConnection", hasConnectionString: true);
             services.TryAddDataProvider(name: "Sql Lite", value: "Sqlite", hasConnectionString: false);
 
+            SqlDialectFactory.RegisterSqlDialect("diagnosticdbconnection", new SqlServerDialect());
```

Comment:
I understand the problem. I will look at your code and see what I can change. Right now I think I could add a custom configuration to accept a dialect connection name. Or also provide a way to hook up a connection wrapper which would not alter the dialect. Would that work?

Are there any other changes we should do to make the module not intrusive at all?